go/teams/sig.go

```diff
@@ -267,20 +287,33 @@ func precheckLinkToPost(ctx context.Context, g *libkb.GlobalContext,
 
 func precheckLinksToPost(ctx context.Context, g *libkb.GlobalContext,
 	sigMultiItems []libkb.SigMultiItem, state *TeamSigChainState, me keybase1.UserVersion) (err error) {
-
 	defer g.CTraceTimed(ctx, "precheckLinksToPost", func() error { return err })()
 
+	// As an optimization, AppendChainLink consumes its state.
+	// We don't consume our state parameter.
+	// So clone state before we pass it along to be consumed.
+	if state != nil {
+		state = state.DeepCopyToPtr()
+	}
+
+	for _, sigItem := range sigMultiItems {
+		state, err = playSigItem(ctx, g, me, state, sigItem)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func playSigItem(ctx context.Context, g *libkb.GlobalContext, me keybase1.UserVersion, state *TeamSigChainState, sigMultiItem libkb.SigMultiItem) (*TeamSigChainState, error) {
```

Comment:
I'd suggest putting `precheck` in the name like `playPrecheckSigItem` so that nobody's tempted to call this for anything else.